### PR TITLE
Alias warn method

### PR DESCRIPTION
**Pull request by rifkinni**
# Changes
- adds an alias `warn` for the `warning` method
- tests the logged level for all methods in the public interface

@simplybusiness/application-tooling

### CODE REVIEW

File: lib/twiglet/logger.rb [hunk 1]
```diff
@@ -36,6 +36,8 @@ def warning(message)
       log(level: 'warning', message: message)
     end
 
+    alias :warn :warning
```
Comment (moltenice) [inline lib/twiglet/logger.rb]: This is genuinely such a cool Ruby feature! The first time I feel like Javascript is missing out on something that Ruby has :P

**Review by moltenice — APPROVED** [review]
Bringing back the 🌮

File: lib/twiglet/logger.rb (hunk 1 above)
Comment (punitcse) [reply lib/twiglet/logger.rb]: Not big deal but I would prefer `alias_method` over `alias`.

File: test/logger_test.rb [hunk 2]
```diff
@@ -182,6 +182,23 @@
...
+  levels = [
+    { method: :debug, level: 'debug' },
+    { method: :info, level: 'info' },
+    { method: :warning, level: 'warning' },
+    { method: :warn, level: 'warning' },
+    { method: :error, level: 'error' }
+  ]
+
+  levels.each do |attrs|
+    it 'should correctly log level' do
```
Comment (punitcse) [inline test/logger_test.rb]: Would it help to add `attrs[:level]` in the test description?

File: lib/twiglet/logger.rb (hunk 1 above)
Comment (sleepyfox) [reply lib/twiglet/logger.rb]: Nope, JS has something very similar: `this.warn = this.warning`, or see the node implementation.

File: test/logger_test.rb [hunk 3]
```diff
@@ -182,6 +182,23 @@
...
+      @logger.public_send(attrs[:method], {message: 'a log message'})
+      actual_log = read_json(@buffer)
+
+      assert_equal attrs[:level], actual_log[:log][:level]
```
Comment (sleepyfox) [inline test/logger_test.rb:198]: I would want to assert that the message came through in each case as well.

File: lib/twiglet/logger.rb (hunk 1 above)
Comment (moltenice) [reply lib/twiglet/logger.rb]: Ah yeah of course! JS still doesn't disappoint 😄

**Review by punitcse — APPROVED** [review]
🍰

**Pull request merged**